src/odemis/acq/test/fastem_test.py

```diff
@@ -216,6 +216,46 @@ def test_calculate_field_indices(self):
 
         self.assertListEqual(expected_indices, field_indices)
 
+    def test_calculate_field_indices_overlap(self):
+        """Check that the correct number of field indices are calculated when there is overlap between the fields."""
+        x_fields = 3
+        y_fields = 2
+        self.multibeam.resolution.value = (6400, 6400)  # don't change
+        res_x, res_y = self.multibeam.resolution.value  # single field size
+        px_size_x, px_size_y = self.multibeam.pixelSize.value
+
+        roa_name = time.strftime("test_megafield_id-%Y-%m-%d-%H-%M-%S")
+
+        field_size_x = res_x * px_size_x
+        field_size_y = res_y * px_size_y
+
+        for overlap in (0, 0.0625, 0.2, 0.5, 0.7):
+            # The coordinates of the ROA in meters.
+            xmin, ymin = (0, 0)
+            # The max field size is the number of fields multiplied with the field size including overlap, plus
+            # the extra overlap added to the end.
```

Comment:
What do you mean with "extra overlap to the end"?

Reply:
The way I understand it is that if we have overlap the fields within a megafield overlap. The last row and column do not overlap with any fields to there right or bottom, therefore they can actually use their full size.
Say you have 3 fields of 8 of these '-' and the overlap is 2, then the total size is 6 * 3 + 2. I tried to visualize it, where the left border of each field is drawn with a | and the right border of each field with a ⁞
|------|--⁞----|--⁞----|--⁞

Reply:
I like the example :)  Please add it!